### PR TITLE
Refine error when opening store in inconsistent state

### PR DESCRIPTION
**Pull request by icristescu**

### CODE REVIEW

File: src/irmin-pack/unix/append_only_file.ml [hunk 1]
```diff
@@ -52,19 +52,31 @@ module Make (Io : Io.S) = struct
...
       as the real offset.
 
-      In case of a crash, if the real offset is larger than the [end_offset], we
-      can safely revert back to [end_offset]. If not, then the store is
-      corrupted, an error is returned. *)
-  let check_consistent_store ~end_offset io =
+      In case of a crash:
+
+      - if the real offset is larger than the [end_offset], we can safely revert
+        back to [end_offset].
```
Comment (icristescu) [inline src/irmin-pack/unix/append_only_file.ml:58]: this is actually not what we discussed with @samoht and @metanivek - we should always take the real_offset and update the control file accordingly. In this way, we ensure that we don't crash if the index points to something between [end_offset] and the real offset. I'll push a fix.

**Comment by icristescu** [comment]
After discussions with @metanivek and @Ngoguey42, I updated the PR - we cannot recover in the case that the real offset on the disk is smaller than the corresponding end_offset in the control file, because it could be that only part of an object was flushed to disk. The end offset specified in the control file is the only one that guarantees that the suffix file is in a consistent state.

File: src/irmin-pack/unix/append_only_file.ml [hunk 2]
```diff
@@ -52,19 +52,25 @@ module Make (Io : Io.S) = struct
...
+  let check_consistent_store ~end_offset ~dead_header_size io =
     let open Result_syntax in
     let* real_offset = Io.read_size io in
-    if real_offset < end_offset then Error `Inconsistent_store else Ok ()
+    let dead_header_size = Int63.of_int dead_header_size in
+    let real_offset_without_header =
+      Int63.Syntax.(real_offset - dead_header_size)
+    in
+    if real_offset_without_header < end_offset then Error `Inconsistent_store
+    else Ok ()
```
Comment (Ngoguey42) [inline src/irmin-pack/unix/append_only_file.ml:67]: Can you print a warning in the `real_offset_without_header > end_offset` case?

Comment (icristescu) [reply src/irmin-pack/unix/append_only_file.ml]: done, thanks

**Review by Ngoguey42 — APPROVED** [review]

**Pull request merged**